### DISCUSSION TITLE
Custom nodes styling

### DISCUSSION THREAD

**Discussion by royassis** [xyflow/xyflow#4126] [Q&A]
Hey

I'm dynamically setting the style of a custom node. 
I'm trying to set the border to be red and a bit bolder. But this is what i'm getting:

![image](https://github.com/xyflow/xyflow/assets/34189329/c59e3730-64f8-4b9a-85da-b9d9153aa0d3)

As you can see it seems like another red square is put on top of the previous box. 
Also the node box sizes are not of fixed size, and are "hugging" the text. 

This is the style im using:
```typescript
const HighlightedStyle = {
  border: "3px solid red"
}
```

Any way to fix these issues? 
Also is it considered better practice to use className and css for styling ?

**Comment by moklick** [comment]
could you show the code of your custom node? it seems that you've defined a style for the inner node div. When you use `node.style` it gets applied to the outer node wrapper.

```html
  <div class="react-flow__node">` <-- `node.style` gets applied here -->
    <div> <-- your custom node -->
```

**Comment by royassis** [comment]
Hey @moklick, you are correct.

```typescript
const CustomNode = ({
  data, selected, isConnectable
}: NodeProps) => {

  return <div style={getNodeStyle(data, selected)}>
    <Handle
      type="target"
      position={Position.Top}
      isConnectable={isConnectable}
    />
    <p>{data.label}</p>

    <Handle
      type="source"
      id="b"
      position={Position.Bottom}
      isConnectable={isConnectable}
    />
  </div>
}
```

So is it possible to change the **outer node wrapper** style from within a custom node ? 
If not do you have a better solution for my use case? I'm trying to style nodes based on their properties.
